quantgen/R/quantile_ensemble.R

```diff
@@ -105,7 +105,7 @@ quantile_ensemble = function(qarr, y, tau, weights=NULL,
   # Add an all 1s matrix to qarr, if we need to
   if (intercept) {
     a = array(NA, dim=c(n,p+1,r))
-    for (k in 1:r) a[,,k] = cbind(rep(1,n), qarr[,,k])
+    for (k in 1:r) a[,,k] = cbind(rep(1,n), `dim<-`(qarr[,,k], apply(qarr, 3, dim)[, 1]))
```

Comment:
Could this be more simply achieved with `qarr[,,k, drop = FALSE]`?

Reply:
I don't think `drop = FALSE` works here as it'll return a full array where we need a column vector:

```{r}
> qarr <- array(rep(1, 12), dim = c(3, 1, 4))
> qarr[,, 3, drop = FALSE]
, , 1

     [,1]
[1,]    1
[2,]    1
[3,]    1

> `dim<-`(qarr[,,3], apply(qarr, 3, dim)[, 1])
     [,1]
[1,]    1
[2,]    1
[3,]    1
```

Reply:
Apologies again @sbfnk! Again I missed this. Please tag me (with @ryantibs) otherwise I'm afraid I won't see the notification when you respond. And we'll wait another several months until I randomly think to check this repo...

Anyway, yikes. That's too and the simpler option doesn't work and thanks for showing me. I'll resolve this conversation now.